src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -85,15 +87,20 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
             return result.Return;
         }
 
-        private async Task DelayUntilFunctionDispatcherInitialized()
+        private async Task DelayUntilFunctionDispatcherInitializedOrShutdown()
         {
             if (_functionDispatcher != null && _functionDispatcher.State == FunctionInvocationDispatcherState.Initializing)
             {
                 _logger.LogDebug($"functionDispatcher state: {_functionDispatcher.State}");
-                await Utility.DelayAsync(WorkerConstants.ProcessStartTimeoutSeconds, WorkerConstants.WorkerReadyCheckPollingIntervalMilliseconds, () =>
+                await Utility.DelayAsync(_functionDispatcher.ErrorEventsThreshold * WorkerConstants.ProcessStartTimeoutSeconds, WorkerConstants.WorkerReadyCheckPollingIntervalMilliseconds, () =>
```

Comment:
Can we make this `_functionDispatcher.ErrorEventsThreshold + 1`? Just to because the code in WorkerFunctionInvoker that kills the running site is meant to be as a last resort, or in case of bugs with function dispatcher's shutdown code.